extensions-core/namespace-lookup/src/main/java/io/druid/query/extraction/namespace/URIExtractionNamespace.java

```diff
@@ -64,40 +64,58 @@
   @JsonProperty
   private final URI uri;
   @JsonProperty
+  private final URI uriPrefix;
+  @JsonProperty
   private final FlatDataParser namespaceParseSpec;
   @JsonProperty
-  private final Period pollPeriod;
+  private final String fileRegex;
   @JsonProperty
-  private final String versionRegex;
+  private final Period pollPeriod;
 
   @JsonCreator
   public URIExtractionNamespace(
       @NotNull @JsonProperty(value = "namespace", required = true)
-      String namespace,
-      @NotNull @JsonProperty(value = "uri", required = true)
-      URI uri,
+          String namespace,
+      @JsonProperty(value = "uri", required = false)
+          URI uri,
+      @JsonProperty(value = "uriPrefix", required = false)
+          URI uriPrefix,
+      @JsonProperty(value = "fileRegex", required = false)
+          String fileRegex,
       @JsonProperty(value = "namespaceParseSpec", required = true)
-      FlatDataParser namespaceParseSpec,
+          FlatDataParser namespaceParseSpec,
       @Min(0) @Nullable @JsonProperty(value = "pollPeriod", required = false)
-      Period pollPeriod,
+          Period pollPeriod,
+      @Deprecated
       @JsonProperty(value = "versionRegex", required = false)
-      String versionRegex
+          String versionRegex
   )
   {
-    if (versionRegex != null) {
+    this.namespace = Preconditions.checkNotNull(namespace, "namespace");
```

Comment:
i am not sure why we need this ? and if so what is the next PR ? why not doing all in one PR ?